test/source/tests/setup.ts

```diff
@@ -497,6 +437,32 @@ AN8G3r5Htj8olot+jm9mIa5XLXWzMNUZgg==
       await gmailPage.notPresent(['@webmail-notification-setup', '@notification-setup-action-open-settings', '@notification-setup-action-dismiss', '@notification-setup-action-close']);
     }));
 
+    ava.default('setup - test adding missing self-signature key issue', testWithBrowser('compatibility', async (t, browser) => {
+      const acctEmail = 'flowcrypt.compatibility@gmail.com';
+      const settingsPage = await browser.newPage(t, TestUrls.extensionSettings(acctEmail));
+      await SettingsPageRecipe.toggleScreen(settingsPage, 'additional');
+      const addKeyPopup = await SettingsPageRecipe.awaitNewPageFrame(settingsPage, '@action-open-add-key-page', ['add_key.htm']);
+      await addKeyPopup.waitAndClick('@source-paste');
+      const key = Config.key('missing.self.signatures');
+      await addKeyPopup.waitAndType('@input-armored-key', key?.armored ?? '');
+      await addKeyPopup.waitAndType('#input_passphrase', key?.passphrase ?? '');
+      await addKeyPopup.waitAndClick('.action_add_private_key', { delay: 1 });
+      await addKeyPopup.waitAll('@input-compatibility-fix-expire-years', { timeout: 30 });
+      await addKeyPopup.selectOption('@input-compatibility-fix-expire-years', '1');
+      await addKeyPopup.waitAndClick('@action-fix-and-import-key');
+      await Util.sleep(1);
+      const myKeyFrame = await SettingsPageRecipe.awaitNewPageFrame(settingsPage, `@action-show-key-2`, ['my_key.htm', 'placement=settings']);
+      await Util.sleep(1);
+      const curDate = new Date(), year = curDate.getFullYear(), month = curDate.getMonth(), date = curDate.getDate();
+      const expirationDate = new Date(year + 1, month, date);
+      // Had to add this because if test runs at 23:59:59 it might cause assertion error
+      // https://github.com/FlowCrypt/flowcrypt-browser/pull/4796#discussion_r1025150001
+      const oneDayAfterExpirationDate = new Date(year + 1, month, date + 1);
+      const expiration = Str.datetimeToDate(Str.fromDate(expirationDate));
+      const oneDayAfterExpiration = Str.datetimeToDate(Str.fromDate(oneDayAfterExpirationDate));
+      expect(await myKeyFrame.read('@content-key-expiration')).to.be.oneOf([expiration, oneDayAfterExpiration]);
+    }));
```

Comment:
Now you're comparing `2023-11-16' (created at 23:59) against ['2023-11-17', '2023-11-18'].
Should be [oneDayBeforeExpectedExpiration, expectedExpiration]